### PR TITLE
Fixes for IMDS authentication in app services

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Fixed spelling of clientid query param for legacy authentication
endpoint.
Removed metadata header from authentication request.
Fixed parsing of expires_on in timestamp format.
Added the endpoint in the authentication failure error message.

Thank you for your contribution to Go-AutoRest! We will triage and review it as soon as we can.

As part of submitting, please make sure you can make the following assertions:
 - [ ] I've tested my changes, adding unit tests if applicable.
 - [ ] I've added Apache 2.0 Headers to the top of any new source files.
 - [ ] I'm submitting this PR to the `dev` branch, except in the case of urgent bug fixes warranting their own release.
 - [ ] If I'm targeting `master`, I've updated [CHANGELOG.md](https://github.com/Azure/go-autorest/blob/master/CHANGELOG.md) to address the changes I'm making.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes:

- https://github.com/Azure/go-autorest/issues/574
- https://github.com/Azure/go-autorest/issues/577

**Comment by jhendrixMSFT** [comment]
Need to add/fix up some more tests for this.

File: autorest/adal/token.go [hunk 1]
```diff
@@ -76,6 +77,9 @@ const (
 
 	// the API version to use for the App Service MSI endpoint
 	appServiceAPIVersion = "2017-09-01"
+
+	// the format for expires_on in UTC (applicable to MSI via legacy ASE only)
+	expiresOnDateFormat = "1/2/2006 15:04:05 PM +00:00"
```
Comment (jhendrixMSFT) [inline autorest/adal/token.go:82]: @catalinaperalta this will need to be forward-ported to `azidentity`

**Review by catalinaperalta — APPROVED** [review]
lgtm

**Pull request merged**